core/src/main/scala/com/databricks/labs/remorph/generators/sql/LogicalPlanGenerator.scala

```diff
@@ -34,12 +34,71 @@ class LogicalPlanGenerator(val expr: ExpressionGenerator, val explicitDistinct:
     case ir.DeleteFromTable(target, None, where, None, None) => delete(ctx, target, where)
     case c: ir.CreateTableCommand => createTable(ctx, c)
     case t: ir.TableSample => tableSample(ctx, t)
+    case a: ir.AlterTableCommand => alterTable(ctx, a)
     case ir.NoopNode => ""
     case ir.UnresolvedCommand(inputText) => s"--${inputText}"
     case null => "" // don't fail transpilation if the plan is null
     case x => throw unknown(x)
   }
 
+  private def alterTable(ctx: GeneratorContext, a: ir.AlterTableCommand): String = {
+    val operation = buildTableAlteration(ctx, a.alterations)
+    s"ALTER TABLE ${a.tableName} $operation"
+  }
+
+  private def buildTableAlteration(ctx: GeneratorContext, alterations: Seq[ir.TableAlteration]): String = {
+    alterations.headOption match {
```

Comment:
it's not clear why do you need a headOption. alterations is a seq, so you have to do:

```
alterations.map {
   case ir.AddColumn(decl) => columnDeclaration(decl)
   case ir.AddConstraint(columnName, constraint) => ...
   ..
}
```